System/Random.hs

```diff
@@ -1002,7 +1005,11 @@ instance Random Double where
   random = randomDouble
   randomM = uniformR (0, 1)
 
-instance UniformRange Double
+instance UniformRange Double where
+  uniformR (l, h) g = do
+    w64 <- uniformWord64 g
+    let x = castWord64ToDouble $ (w64 `shiftR` 12) .|. 0x3ff0000000000000
```

Comment:
Switching to `unsafeShiftR` would give a bit of a boost (Same in the `Float` instance)